blueprints/ember-electron/index.js

```diff
@@ -24,96 +21,143 @@ class EmberElectronBlueprint extends Blueprint {
+    // n.b. addPackageToProject does not let us save prod deps, so we task
+    let npmInstall = this.taskFor('npm-install');
+
+    logger.startProgress('Installing electron build tools');
+
+    return efImport({
+      updateScripts: false,
+      outDir: 'electron-out',
+    })
+      .then(() => npmInstall.run({
+        saveDev: true,
+        verbose: false,
+        packages: ['devtron@^1.4.0'],
+      }))
+      .then(() => npmInstall.run({
+        save: true,
+        verbose: false,
+        packages: ['electron-protocol-serve@^1.3.0'],
+      }))
+      .then(() => logger.message('Installed electron build tools'));
   }
 
-  _createResourcesDirectories() {
-    const platforms = ['', 'darwin', 'linux', 'win32'];
-    let promises = platforms.map((platform) => {
-      let gitKeepPath = path.join(this.options.project.root, 'ember-electron', `resources${platform ? '-' : ''}${platform}`, '.gitkeep');
-      return ensureFile(gitKeepPath);
-    });
+  _createResourcesDirectories(logger) {
+    const ensureFile = denodeify(fs.ensureFile);
 
-    return all(promises);
+    logger.startProgress('Creating ember-electron resource dirs');
+
+    let rootDir = this.options.project.root;
+    let ensureResourceDirGitKeeps = [
+      'resources',
+      'resources-darwin',
+      'resources-linux',
+      'resources-win32',
+    ]
+      .map((dirName) => path.join(rootDir, 'ember-electron', dirName))
+      .map((dirPath) => ensureFile(path.join(dirPath, '.gitkeep')));
+
+    return all(ensureResourceDirGitKeeps)
+      .then(() => logger.message('Created ember-electron resource dirs'));
   }
 
-  _ensurePackageJsonConfiguration() {
+  _ensureForgeConfig(logger) {
+    const readJson = denodeify(fs.readJson);
+    const writeFile = denodeify(fs.writeFile);
+    const writeJson = denodeify(fs.writeJson);
+
     let packageJsonPath = path.join(this.project.root, 'package.json');
-    // We can't use this.project.pkg because we modified package.json by
-    // installing dependencies, and that won't be reflected in this.project.pkg
+    let forgeConfigPath = './ember-electron/.electron-forge';
+
+    if (this.hasForgeConfig) {
+      return;
+    }
+
+    logger.startProgress('Extracting ember-electron forge config');
+
     return readJson(packageJsonPath)
       .then((packageJson) => {
-        packageJson.config = packageJson.config || {};
-        packageJson.config.forge = './ember-electron/.electron-forge';
+        let forgeConfig = packageJson.config.forge;
+
+        if (typeof packageJson.config.forge === 'string') {
+          return;
+        }
+
+        forgeConfig = JSON.stringify(forgeConfig, null, 2);
+
+        delete packageJson.config.forge;
+
+        if (Object.keys(packageJson.config).length === 0) {
+          delete packageJson.config;
+        }
 
-        return writeJson(packageJsonPath, packageJson, { spaces: 2 });
+        return all([
+          writeFile(forgeConfigPath, `module.exports = ${forgeConfig}`),
```

Comment:
We should end the file content with a `;` to keep consisting with generally accepted default linting standards.